File: src/main/java/fr/lewon/bot/runner/bot/props/BotPropertyStore.java
package fr.lewon.bot.runner.bot.props;

import java.util.HashMap;

public class BotPropertyStore extends HashMap<BotPropertyDescriptor, Object> {

    private static final long serialVersionUID = 8101778657485578888L;

    public BotPropertyStore() {
    }

    public BotPropertyStore(BotPropertyStore botPropertyStore) {
        super(botPropertyStore);
    }

    public Object getByKey(String key) {
        return this.get(this.keySet().stream()
                .filter(p -> p.getKey().equals(key))
                .findFirst().orElse(null));
    }

    public void addProperty(BotPropertyDescriptor key, Object value) {
        this.put(key, value);
    }

}

File: src/main/kotlin/fr/lewon/bot/runner/bot/props/BotPropertyStore.kt
package fr.lewon.bot.runner.bot.props

import java.util.*

class BotPropertyStore : HashMap<BotPropertyDescriptor, Any?>() {

    fun getByKey(key: String): Any? {
        return this[this.keys.stream()
            .filter { p -> p.key == key }
            .findFirst().orElse(null)]
    }

    fun containsByKey(key: String): Boolean {
        return this.keys.stream()
            .filter { p -> p.key == key }
            .findFirst().orElse(null) != null;
    }

    companion object {
        private const val serialVersionUID = 8101778657485578888L
    }

}
